migrations/20210612_initial.sql

```diff
@@ -0,0 +1,104 @@
+CREATE TABLE profiles (
+  profile_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL,
+  properties TEXT NOT NULL -- JSON with user or contact profile
+);
+
+CREATE TABLE known_servers(
+  host TEXT NOT NULL,
+  port TEXT NOT NULL,
+  key_hash BLOB,
+  PRIMARY KEY (host, port)
+) WITHOUT ROWID;
+
+CREATE TABLE contacts (
+  contact_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL, -- display name set by local user
+  properties TEXT NOT NULL, -- JSON set by local user
+  profile_id INTEGER REFERENCES profiles, -- sent by remote contact
+  contact_status TEXT NOT NULL DEFAULT ''
+);
+
+CREATE TABLE contact_connections (
+  connection_id BLOB NOT NULL PRIMARY KEY,
+  contact_id INTEGER NOT NULL REFERENCES contacts ON DELETE RESTRICT,
+  conn_level INTEGER NOT NULL DEFAULT 0,
+  via_conn BLOB REFERENCES contact_connections (connection_id),
+  conn_status TEXT NOT NULL DEFAULT ''
+) WITHOUT ROWID;
+
+CREATE TABLE contact_invitations (
+  invitation_id BLOB NOT NULL PRIMARY KEY,
+  contact_id INTEGER NOT NULL REFERENCES contacts ON DELETE RESTRICT,
+  invitation_status TEXT NOT NULL DEFAULT ''
+) WITHOUT ROWID;
+
+CREATE TABLE groups (
+  group_id INTEGER NOT NULL PRIMARY KEY, -- local group ID
+  display_name TEXT NOT NULL, -- shared
```

Comment:
why is `display_name` shared here but local on `contacts`? does a group even need a local display name? maybe it could just be part of shared `properties`?

also `properties` -> `shared_properties` / `external_properties`? 1st is it's just for explicitness since we use the same field name for different purposes in different tables, 2nd is we might need `local` / `internal_properties` for groups as well - e.g. whether group is muted or not

Reply:
> why is display_name shared here but local on contacts?

contacts have both local and shared (this one in profiles).

> does a group even need a local display name?

I don't think so. I am actually now thinking there should be a history of updates in the same way as for messages linked to events, both for groups and contacts profiles... Although we could add local group name to support anonymous groups without shared metadata, but there are shared member IDs still...

> maybe it could just be part of shared properties

I was planning to have it all in JSON in the protocol, but thought it's better to extract the main field into a separate record field - to simplify search, for example.

> properties -> shared_properties

will think... Maybe local_ instead...

Reply:
> whether group is muted or not

not sure what you mean

Reply:
you commented that it's a shared `properties` - I was saying group can additionally have `local_properties` in user's app - like whether to mute (silence notifications) the group for him or not, maybe some others